### PR TITLE
mopac: init at 22.0.6

### PR DESCRIPTION
**Pull request by sheepforce**
###### Description of changes

Addition of the semiempirical quantum chemistry programme Mopac. Part of the ongoing effort to upstream packages from NixOS-QChem, see https://github.com/markuskowa/NixOS-QChem/issues/146

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: pkgs/applications/science/chemistry/mopac/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+, lib
+, gfortran
+, fetchFromGitHub
+, cmake
+, blas
+, lapack
+, python3Packages
+}:
+
+assert blas.isILP64 == lapack.isILP64;
```
Comment (markuskowa) [inline pkgs/applications/science/chemistry/mopac/default.nix:11]: Does the package automatically detect ILP64 (I do not see that here)? Maybe we should also assert `isILP64 == false`.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -9116,6 +9116,8 @@ with pkgs;
 
   mob = callPackage ../applications/misc/mob {};
 
+  mopac = callPackage ../applications/science/chemistry/mopac { };
```
Comment (markuskowa) [inline pkgs/top-level/all-packages.nix:9119]: I think there is a "chemistry" section  in the top level file.

File: pkgs/applications/science/chemistry/mopac/default.nix (hunk 1 above)
Comment (sheepforce) [reply pkgs/applications/science/chemistry/mopac/default.nix:11]: It actually seems to do, yes. We already figured this out in NixOS-QChem: https://github.com/markuskowa/NixOS-QChem/pull/217#discussion_r989067288

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (sheepforce) [reply pkgs/top-level/all-packages.nix]: Uh, nice. Completely missed it. `all-packages.nix` is crazy :)

File: pkgs/applications/science/chemistry/mopac/default.nix (hunk 1 above)
Comment (markuskowa) [reply pkgs/applications/science/chemistry/mopac/default.nix:11]: Oh, ok, never mind then.

**Pull request merged**